web/rest/user/config/api/raw/provider.yml

```
@@ -142,8 +140,11 @@ Ivoz\Provider\Domain\Model\Voicemail\Voicemail:
     access_control: '"ROLE_COMPANY_USER" in roles'
     read_access_control:
       ROLE_COMPANY_USER:
-        user:
-          eq: "user.getId()"
+        - or:
+          - id:
+              in: "voicemailRepository.getVoicemailsIdsByUser(user)"
+          - id:
+              in: "voicemailRelUserRepository.getVoicemailIdsByUser(user)"
```

```suggestion
        - or:
          - user:
              eq: "user.getId()"
          - id:
              in: "voicemailRelUserRepository.getVoicemailIdsByUser(user)"
```

We don't need an extra query to check if the voicemail belongs to the user.